### PR TITLE
Create mergeSort.js

### PR DESCRIPTION
**Pull request by jorgegonzalez**

### CODE REVIEW

File: utilities/javascript/mergeSort.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      merged.push(arr2[j]);
+      j++;
+    } else {
+      merged.push(arr1[i]);
+      i++;
+      j++;
+    }
+  }
+
+  while (i < arr1.length) {
```
Comment (yangshun) [inline utilities/javascript/mergeSort.js]: Instead of:
```
 +  while (i < arr1.length) {
 +    merged.push(arr1[i]);
 +    i++;
 +  }
 +  while (j < arr2.length) {
 +    merged.push(arr2[j]);
 +    j++;
 +  }

```
This is shorter:

```
merged.push(...arr1.slice(i), ...arr2.slice(j));
```

File: utilities/javascript/mergeSort.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  let j = 0;
+
+  while (i < arr1.length && j < arr2.length) {
+    if (arr1[i] < arr2[j]) {
+      merged.push(arr1[i]);
+      i++;
+    } else if (arr2[j] < arr1[i]) {
+      merged.push(arr2[j]);
+      j++;
+    } else {
```
Comment (yangshun) [inline utilities/javascript/mergeSort.js]: This part looks odd. You're incrementing two indices but only pushing one value. That will result in a missing value in the final array. This case is where both values are equal. Few ways to fix it:

1. Push both `arr1[i]` and `arr2[j]` 
2. Change `arr1[i] < arr2[j]` above to `arr1[i] <= arr2[j]`. 

I prefer the latter as fewer lines of code are required.

File: utilities/javascript/mergeSort.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  return merged;
+}
+
+const arr1 = [7, 2, 4, 3, 1, 2];
+const arr2 = [10, 9, 8, 7, 6, 5, 4, 3, 2, 1];
+const arr3 = [98322, 3242, 876, -234, 34, 12331];
+const arr4 = [1, 2, 3, 4, 5, 0];
+
+console.log(mergeSort(arr1));
```
Comment (yangshun) [inline utilities/javascript/mergeSort.js]: Could you add the result and compare using `===` like in `binarySearch.js`?

Also, please add a test for duplicate values and corner cases with 0, 1, 2 elements in the array.

**Comment by jorgegonzalez** [comment]
I was going to make those changes today as I ran out of time yesterday, but it looks like you already did.

Thank you for letting me contribute to this project, by the way.

**Comment by yangshun** [comment]
Hey @jorgegonzalez, no problem! The changes were non-trivial as I had to write a `deepEqual` function, so I decided to do it for you, hope you don't mind!
